### DISCUSSION TITLE
Supplying a root CA inside docker container

### DISCUSSION THREAD

**Discussion by DSc5** [directus/directus#9919] [Feature Requests]
Hello all,

LDAP authentication was a great feature. I like to use it and therefore thank you first.

Currently I have the problem of not getting LDAPS to work. Just LDAP is working fine.
Directus is currently running at version 9.0.0 inside a docker container.

When i set LDAP to LDAPS, i always get timeout errors. The main problem is, that the application inside my docker container does not know my local host certificates. I need to import my root ca in order to get the secured version of LDAP to work.

Thank you

**Comment by DSc5** [comment]
Is there a possibiliy to import my root ca into the application / docker container?

**Reply by rijkvanzanten** [reply]
You can create a custom image using the directus/directus one as a base, and then COPY in the CA file in build, or maybe use a volume in the container?

**Reply by DSc5** [reply]
That was a great idea which i was already trying.
When i docker exec sh into the directus container, i am possible now to wget some ressources from my local network via ssl. So the certificate trusted and this is working.  But LDAP with TLS / SSL not.

Now I guess that the LDAP implementation in the application must provide TLS/SSL extra.

**Reply by rijkvanzanten** [reply]
Not sure about that! @aidenfoxx any ideas around LDAP vs LDAPS ?

**Reply by aidenfoxx** [reply]
The library we use accepts node tls options which can be passed via `AUTH_<PROVIDER>_CLIENT__TLS_OPTIONS` http://ldapjs.org/client.html

**Reply by aidenfoxx** [reply]
If the container is correctly configured, however, there should be no need to pass the extra CA options

**Reply by DSc5** [reply]
@aidenfoxx 
Thank you very much for your reply! 
I am familiar with ldap, but im having trouble to figure out how the env has to be defined.
I tried something like AUTH_LDAP_CLIENT__TLS_OPTIONS="{rejectUnauthorized: false}". But I doubt that this is correct.

Thanks in advance

**Reply by rfehling-mittwald** [reply]
Guess you should try:

AUTH_LDAP_CLIENT__TLS_OPTIONS="json:{rejectUnauthorized: false}"

[(These line would set this to empty object)](https://github.com/directus/directus/blob/main/api/src/auth/drivers/ldap.ts#L69)

But still have tls issues. Hard to setup certificates via envs with directus helm charts https://github.com/directus-community/helm-chart :/

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
